src/webclients/components/sdfv/utils/helpers.ts

```diff
@@ -97,44 +97,50 @@ export function findJsonSDFGElementByUUID(cfgList: CFGListType, uuid: string): [
     return [undefined, rootSDFG];
 }
 
+function recursiveDoForScopeChildren(
+    element: SDFGNode, action: CallableFunction
+): void {
+    const stateId = element.parent_id;
+    if (element instanceof EntryNode && stateId !== null &&
+        !element.attributes().is_collapsed) {
+        const state = element.cfg?.nodes[stateId] as JsonSDFGState;
+        if (state?.scope_dict[element.id] !== undefined) {
+            const stateGraph = element.parentElem?.data.graph;
+            if (!stateGraph) {
+                throw Error(
+                    'State graph missing on node' + element.label()
+                );
+            }
+            for (const nId of state.scope_dict[element.id]) {
+                const nd = stateGraph.node(nId);
+                action(nd);
+                recursiveDoForScopeChildren(nd, action);
+            }
+        }
+    }
+}
+
 /**
  * Perform an action for each element in an array given by their uuids.
  */
 export function doForAllUUIDs(
     uuids: string[], action: CallableFunction,
-    applyToScopeChildren: boolean = false
+    applyToScopeChildren: boolean = false, applyToNone: boolean = false
```

Comment:
Apply to null/undefined? I know it’s from Python but it’s strange to see here. Also some doc would be nice.

Reply:
Fixed